test/functional/normalization/json_format_test.go

```diff
@@ -0,0 +1,145 @@
+package normalization
+
+import (
+	"encoding/json"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test/functional"
+	"github.com/openshift/cluster-logging-operator/test/helpers/types"
+	"github.com/openshift/cluster-logging-operator/test/matchers"
+	"strings"
+)
+
+var _ = Describe("[Functional][Normalization][Json] Parse json format log", func() {
+
+	var (
+		framework *functional.FluentdFunctionalFramework
+		// json message
+		jsonMsg = "{\\\"name\\\":\\\"fred\\\",\\\"home\\\":\\\"bedrock\\\"}"
+		// Template expected as output Log
+	)
+
+	BeforeEach(func() {
+
+		framework = functional.NewFluentdFunctionalFramework()
+		functional.NewClusterLogForwarderBuilder(framework.Forwarder).
+			FromInput(logging.InputNameApplication).
+			ToFluentForwardOutput()
+	})
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	Context("when parsing json logs as un-structured", func() {
+		It("should be logged as a message", func() {
+			Expect(framework.Deploy()).To(BeNil())
+			Expect(framework.WritesJsonApplicationLogs(jsonMsg, 1)).To(BeNil())
+
+			raw, err := framework.ReadApplicationLogsFrom("fluentForward")
+			Expect(err).To(BeNil(), "Expected no errors reading the logs")
+			Expect(raw).To(Not(BeEmpty()))
+
+			// Parse log line
+			logs, err := types.ParseLogs(raw)
+			Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+			// Compare to expected template
+			outputLogTemplate := functional.NewLogTemplate()
+			outputLogTemplate.Message = strings.Replace(jsonMsg, "\\", "", -1)
+			outputTestLog := logs[0]
+			Expect(outputTestLog).To(matchers.FitLogFormatTemplate(outputLogTemplate))
+		})
+	})
+
+	Context("when parsing json logs as structured - using current plugin implementation - option 1", func() {
+		It("should be logged as a message", func() {
+
+			framework.SetFluentConfigFileName("fluentd_structured_configuration_option1.txt")
```

Comment:
These configuration should come from logic to produce the config using the generator.  This test in particular should be added to the final implementation.